plugin/src/main/java/net/dzikoysk/funnyguilds/guild/placeholders/GuildPlaceholdersService.java

```diff
@@ -158,11 +174,118 @@ public static GuildPlaceholders createAlliesEnemiesPlaceholders(FunnyGuilds plug
                         entity -> messages.get(entity, config -> config.enemiesNoValue));
     }
 
+    public static GuildPlaceholders createMemberPlaceholders(FunnyGuilds plugin) {
+        PluginConfiguration config = plugin.getPluginConfiguration();
+        MessageService messages = plugin.getMessageService();
+        
+        // Create cache for member list priorities to avoid repeated permission lookups
+        // Cache expires after 1 minute to ensure priorities stay reasonably fresh
+        LoadingCache<MemberPriorityKey, Integer> priorityCache = Caffeine.newBuilder()
+                .expireAfterWrite(1, TimeUnit.MINUTES)
+                .build(key -> {
+                    Option<Guild> guildOption = plugin.getGuildManager().findByUuid(key.guildUuid);
+                    if (guildOption.isEmpty()) {
+                        return DEFAULT_MEMBER_PRIORITY;
+                    }
+                    
+                    Option<User> userOption = plugin.getUserManager().findByUuid(key.userUuid);
+                    if (userOption.isEmpty()) {
+                        return DEFAULT_MEMBER_PRIORITY;
+                    }
+                    
+                    return plugin.getGuildPermissionChecker()
```

Comment:
Extract user manager, guild manager and permission checker to variables to reuse it, also permission checker should use getPermissionValue to get Option (which you want to use) not Result